File: 1615-range-sum-of-sorted-subarray-sums/1615-range-sum-of-sorted-subarray-sums.py
class Solution:
    def rangeSum(self, nums: List[int], n: int, left: int, right: int) -> int:
        
        val = []
        for i in range(len(nums)):
            count = 0
            for j in range(i,len(nums)):
                count += nums[j]
                val.append(count)
        mo=10**9+7
        val.sort()
        return sum(val[left-1 :right])%mo

File: 1615-range-sum-of-sorted-subarray-sums/1615-range-sum-of-sorted-subarray-sums.js
/**
 * @param {number[]} nums
 * @param {number} n
 * @param {number} left
 * @param {number} right
 * @return {number}
 */
var rangeSum = function(nums, n, left, right) {
    const list=[]
    for (let i =0;i<n;i++){
        let sum = 0
        for (let j=i;j<n;j++){
            sum+=nums[j]
            list.push(sum)
        }
    }
    list.sort((a,b)=>a-b)
    const mod =10**9+7
    return list.slice(left-1,right)
        .reduce((prev,curr)=>(prev+curr) %mod,0)
};